xinference/core/tests/test_restart_supervisor.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2022-2023 XProbe Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import xoscar as xo
+
+from ...core.supervisor import SupervisorActor
+
+
+# test restart supervisor
+@pytest.mark.asyncio
+async def test_restart_supervisor():
+    from ...deploy.supervisor import run_in_subprocess as supervisor_run_in_subprocess
+    from ...deploy.worker import main as worker_run_in_subprocess
+
+    # start supervisor
+    supervisor_address = "localhost:19034"
```

Comment:
There is a util `from xoscar.utils import get_next_port` can be used to gen a random port.